### ISSUE TITLE
Rules override is reversed on extends

### ISSUE THREAD

**Issue by alias-mac** [ember-template-lint/ember-template-lint#1417]
When defining multiple rules to extend from, the overrides seem to be reversed.

For example, when defining the following:
```js
extends: ["octane", "some-plugin:recommended"]
```
we would expect that rules defined in `some-plugin:recommended` to override any rules (with same name/scope) from 'octane'.

Instead we need to define `extends` reversed for those overrides:
```js
extends: ["some-plugin:recommended", "octane"]
```

Created a quick codesandbox with a reproducible example with some random rule:
https://codesandbox.io/s/eager-morse-t819p?file=/.template-lintrc.js

Running `yarn lint:hbs` in that sandbox passes, when it should fail because the `octane` rule should override the previous defined rules.

While this might be considered a breaking change (in case someone was using it already with "overrides" first) it doesn't work like eslint `extends` and thus seems to be confusing/unexpected.

**Comment by rwjblue** [comment]
Thank you for reporting this! IMHO, it is **absolutely** a bug, but I worry that fixing it will _effectively_ be a breaking change so we have to be prepared for that (and release appropriately).

**Comment by Turbo87** [comment]
resolved by https://github.com/ember-template-lint/ember-template-lint/pull/1513
